native/src/modules/app/components/NavigationTypes.js

```diff
@@ -0,0 +1,109 @@
+
+export type NewsRouteType = 'News'
+export const NEWS_ROUTE: NewsRouteType = 'News'
+
+export type DisclaimerRouteType = 'Disclaimer'
+export const DISCLAIMER_ROUTE: DisclaimerRouteType = 'Disclaimer'
+
+export type OffersRouteType = 'Offers'
+export const OFFERS_ROUTE: OffersRouteType = 'Offers'
+
+export type ExternalOfferRouteType = 'ExternalOffer'
+export const EXTERNAL_OFFER_ROUTE: ExternalOfferRouteType = 'ExternalOffer'
+
+export type SprungbrettOfferRouteType = 'SprungbrettOffer'
+export const SPRUNGBRETT_OFFER_ROUTE: SprungbrettOfferRouteType = 'SprungbrettOffer'
+
+export type WohnenOfferRouteType = 'WohnenOffer'
+export const WOHNEN_OFFER_ROUTE: WohnenOfferRouteType = 'WohnenOffer'
+
+export type SettingsRouteType = 'Settings'
+export const SETTINGS_ROUTE: SettingsRouteType = 'Settings'
+
+export type SearchModalRouteType = 'SearchModal'
+export const SEARCH_MODAL_ROUTE: SearchModalRouteType = 'SearchModal'
+
+export type ChangeLanguageModalRouteType = 'ChangeLanguageModal'
+export const CHANGE_LANGUAGE_MODAL_ROUTE: ChangeLanguageModalRouteType = 'ChangeLanguageModal'
+
+export type PdfViewModalRouteType = 'PDFViewModal'
+export const PDF_VIEW_MODAL_ROUTE: PdfViewModalRouteType = 'PDFViewModal'
+
+export type ImageViewModalRouteType = 'ImageViewModal'
+export const IMAGE_VIEW_MODAL_ROUTE: ImageViewModalRouteType = 'ImageViewModal'
+
+export type FeedbackModalRouteType = 'FeedbackModal'
+export const FEEDBACK_MODAL_ROUTE: FeedbackModalRouteType = 'FeedbackModal'
+
+export type RoutesType = IntroRouteType
+  | LandingRouteType
+  | DashboardRouteType
+  | CategoriesRouteType
+  | PoisRouteType
+  | EventsRouteType
+  | NewsRouteType
+  | DisclaimerRouteType
+  | OffersRouteType
+  | ExternalOfferRouteType
+  | SprungbrettOfferRouteType
+  | WohnenOfferRouteType
+  | SettingsRouteType
+  | SearchModalRouteType
+  | ChangeLanguageModalRouteType
+  | PdfViewModalRouteType
+  | ImageViewModalRouteType
+  | FeedbackModalRouteType
+
+export type RoutesParamsType = {|
+  Intro: void,
+  Landing: void,
+  Dashboard: void,
+  Categories: void,
+  Pois: void,
+  Events: void,
+  News: void,
+  Disclaimer: void,
+  Offers: {| cityCode: string, sharePath: string |},
+  ExternalOffer: {| url: string, postData: ?Map<string, string> |},
+  SprungbrettOffer: {| city: string, offers: Array<OfferModel> |},
+  WohnenOffer: {| offerHash: ?string, city: string, offers: Array<OfferModel> |},
+  Settings: void,
+  SearchModal: void,
+  ChangeLanguageModal: {|
+    currentLanguage: string,
+    previousKey: string,
+    cityCode: string,
+    languages: Array<LanguageModel>,
+    availableLanguages: Array<string>
+  |},
+  PDFViewModal: {| url: string, shareUrl?: string |},
+  ImageViewModal: {| url: string, shareUrl?: string |},
+  FeedbackModal: FeedbackInformationType
+|}
+
+export type RoutePropType<T: RoutesType> = RouteProp<RoutesParamsType, T>
+export type NavigationPropType<T: RoutesType> = StackNavigationProp<RoutesParamsType, T>
```

Comment:
Can you shortly explain again whats the difference between the `RouteProp` and `StackNavigationProp` here?

Reply:
RouteProp contains the state of the react-navigation router, e.g.: all routes in the stack, the name of the current route as well as params (like cityCode, languageCode, categoriesPath)
StackNavigationProp contains methods to change this state, e.g. navigate, replace, dispatch,...